### PR TITLE
TW-874: Refactor dapps screen, add integrate element for Text to Nft

### PR DESCRIPTION
**Pull request by svjatoslavtt**

### CODE REVIEW

File: src/screens/d-apps/integrated-element/integrated-element.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  const styles = useIntegratedElementStyles();
+
+  const handleNavigate = () => navigateFn();
+
+  const themeClasses = {
+    primary: {
+      background: styles.backgroundPrimary,
+      iconBackground: styles.iconBackgroundPrimary
+    }
+  };
```
[the remark this answers is not in the document]
Comment (svjatoslavtt) [reply src/screens/d-apps/integrated-element/integrated-element.tsx:44]: for what?

**Review by herkoss — APPROVED** [review]
GJ!

File: src/screens/d-apps/d-apps.tsx [hunk 2]
```diff
@@ -123,8 +123,8 @@ export const DApps = () => {
         keyExtractor={keyExtractor}
         getItemLayout={getItemLayout}
         numColumns={2}
```
[the remark this answers is not in the document]
Comment (svjatoslavtt) [reply src/screens/d-apps/d-apps.tsx:125]: `horizontal` property has `false` value by default

**Review by herkoss — DISMISSED** [review]

File: src/screens/d-apps/d-apps.tsx [hunk 3]
```diff
@@ -74,49 +77,50 @@ export const DApps = () => {
     return dAppsList;
   }, [searchQuery, dAppsList]);
 
+  const handleLayout = useCallback(({ nativeEvent }: LayoutChangeEvent) => {
+    setLayoutWidth(nativeEvent.layout.width || 1);
+  }, []);
+
+  const elementWidth = useMemo(() => (layoutWidth - gridSize) / 2, [layoutWidth]);
```
[the remark this answers is not in the document]
Comment (svjatoslavtt) [reply src/screens/d-apps/d-apps.tsx:84]: I think it isn't clear to understand what is 48 number in that case

File: src/screens/d-apps/d-apps.tsx [hunk 4]
```diff
@@ -36,38 +36,29 @@ const gridSize = formatSize(48);
...
-      tabletMode
-        ? ['Other DApps are third-party websites. They should be used at your own risk.']
-        : ['Other DApps are third-party websites.', 'They should be used at your own risk.'],
-    [tabletMode]
-  );
+  useEffect(() => {
+    if (partnersPromotionEnabled && !isEnabledAdsBanner) {
+      dispatch(loadPartnersPromoActions.submit(OptimalPromotionAdType.TwMobile));
+    }
+  }, [partnersPromotionEnabled, isEnabledAdsBanner]);
```
[the remark this answers is not in the document]
Comment (svjatoslavtt) [reply src/screens/d-apps/d-apps.tsx:61]: This point is not for this PR. I changed the order of the code but didn't write this logic.

**Review by herkoss — APPROVED** [review]

**Review by vladvn1905 — APPROVED** [review]
ios done build: 1.15_869

**Review by IrynaKhyzhynska — APPROVED** [review]
android done build: 1.15_869

**Pull request merged**